app/Jobs/ProcessNewVideo.php

```diff
@@ -244,6 +244,29 @@ protected function generateVideoThumbnail($path, $time, $width, $height, $format
             ]);
     }
 
+    /**
+     * Generate a thumbnail from the video at the specified time.
+     *
+     * @param string $path Path to the video file.
+     * @param float $time Time for the thumbnail in seconds.
+     * @param int $width Width of the thumbnail.
+     * @param int $height Height of the thumbnail.
+     *
+     * @return string Vips image buffer string.
+     */
+    protected function generateVideoThumbnailForSprites($path, $time, $width, $height)
+    {
+        // Cache the video instance.
+        if (!isset($this->ffmpegVideo)) {
+            $this->ffmpegVideo = FFMpeg::create()->open($path);
+        }
+
+        $buffer = $this->ffmpegVideo->frame(TimeCode::fromSeconds($time))
+            ->save(null, false, true);
+
+        return VipsImage::thumbnail_buffer($buffer, $width, ['height' => $height]);
+    }
+
```

Comment:
Using `generateVideoThumbnail()` causes the test function testHandleThumbnails() to fail. This is because the line` $this->assertEquals([0.5, 5, 9.5], $job->times)`; relies on the array` times`, which gets overwritten or has new elements added each time the `generateVideoThumbnail()` function is called. For a duration of 10 seconds, the array `times `has a length of 10

Reply:
Then you can change the assertion to:
```php
$this->assertEquals([0.5, 5, 9.5], array_slice($job->times, 0, 3));
```

Reply:
of course. I just didn't want to touch the other test functions. Another issue arises when the `generateVideoThumbnail()` function is invoked within the `foreach` loop, specifically in the snippet: `foreach ($times as $index => $time) {
    $buffer = $this->generateVideoThumbnail($path, $time, $width, $height, $format);
    $disk->put("{$fragment}/{$index}.{$format}", $buffer);
}`
The problem stems from the length of the `$times` array being 2 (e.g., [0.5, 1]) when duration is 1s, causing two additional frames/thumbnails to be generated. This leads to a failed assertion at line 80, where the test checks `$this->assertEquals(5, $job->frames);`. The expected number of frames is 5, but due to the extra iterations caused by the length of `$times`, the actual value of `$job->frames `deviates from the expected result.

Reply:
IMO it's ok to 1) modify the old test to take the additional number of sprites into account and 2) modify the new test (line 80) to take the additional number of regular thumbnails into account. While this makes the tests less strict (e.g. there could be +1 thumbnail and -1 sprite and the test would still pass), you could add more "strictness" by comparing the exact times for the sprites, too (as in line 27). This way you can make sure that only thumbnails at the expected times are extracted.